### PR TITLE
Addressing issue #589: Adding alpha params to PCA

### PR DESCRIPTION
**Pull request by naresh-bachwani**
The following changes were made to PCA referenced #589 :
1. Added new parameter alpha to `pca.py `to specify the opacity.
2. Modified `test_pca.py `

The result is attached below:
Screenshot (144)

### CODE REVIEW

**Comment by lwgray** [comment]
@naresh-bachwani This PR still has changes for residuals.

**Comment by lwgray** [comment]
@naresh-bachwani you have changes for both residuals and PCA in this PR.  This needs to be rectified before we move forward

**Comment by naresh-bachwani** [comment]
Hello @lwgray,
Sorry for this error. I'm new to OpenSource, so please guide me towards what needs to be done in order to rectify this. I have made the changes related to residuals in develop branch and adding a new branch for PCA won't work! So should I close this PR until the previous one is merged, if not then please guide me towards what needs to be done!

**Comment by lwgray** [comment]
@naresh-bachwani I am inclined to say to wait until merging #806.  Then once your develop is up-to-date, you should create a feature-branch off of develop for the PCA alpha work.

**Comment by lwgray** [comment]
I can still make comments here while we wait then you can apply them in the new PR

File: tests/test_features/test_pca.py [hunk 1]
```diff
@@ -164,7 +164,7 @@ def test_biplot_3d(self):
         pca_array = visualizer.transform(self.dataset.X)
 
         # Image comparison tests
-        self.assert_images_similar(visualizer)
+        self.assert_images_similar(visualizer,tol=10)
 
         # Assert PCA transformation occurred successfully
         assert pca_array.shape == (self.dataset.X.shape[0], 3)
```
Comment (lwgray) [inline tests/test_features/test_pca.py]: We might want to add a test for alpha.  It could be similar to the alpha test found in residuals.py

**Comment by naresh-bachwani** [comment]
OK. That will be great

File: yellowbrick/features/pca.py [hunk 2]
```diff
@@ -118,7 +119,7 @@ def __init__(self,
             [('scale', StandardScaler(with_std=self.scale)),
              ('pca', PCA(self.proj_dim, random_state=random_state))]
         )
-
+        self.alpha=alpha
```
Comment (lwgray) [inline yellowbrick/features/pca.py]: This is exactly right.  Additionally, We should include alpha parameter in the docstring above

File: yellowbrick/features/pca.py [hunk 3]
```diff
@@ -154,7 +155,7 @@ def transform(self, X, y=None, **kwargs):
     def draw(self, **kwargs):
         X = self.pca_features_
         if self.proj_dim == 2:
-            self.ax.scatter(X[:, 0], X[:, 1], c=self.color, cmap=self.colormap)
+            self.ax.scatter(X[:, 0], X[:, 1], c=self.color, cmap=self.colormap,alpha=self.alpha)
```
Comment (lwgray) [inline yellowbrick/features/pca.py]: place a space after the comma right before alpha

File: yellowbrick/features/pca.py [hunk 4]
```diff
@@ -177,7 +178,7 @@ def draw(self, **kwargs):
             self.fig = plt.figure()
             self.ax = self.fig.add_subplot(111, projection='3d')
             self.ax.scatter(X[:, 0], X[:, 1], X[:, 2],
-                            c=self.color, cmap=self.colormap)
+                            c=self.color, cmap=self.colormap,alpha=self.alpha)
```
Comment (lwgray) [inline yellowbrick/features/pca.py]: place a space after the comma right before alpha

File: yellowbrick/features/pca.py [hunk 5]
```diff
@@ -215,7 +216,7 @@ def finalize(self, **kwargs):
 
 def pca_decomposition(X, y=None, ax=None, features=None, scale=True,
                       proj_dim=2, proj_features=False, color=None,
-                      colormap=palettes.DEFAULT_SEQUENCE,
+                      colormap=palettes.DEFAULT_SEQUENCE,alpha=1,
```
Comment (lwgray) [inline yellowbrick/features/pca.py]: place a space after the comma right before alpha

Comment (naresh-bachwani) [reply yellowbrick/features/pca.py]: OK. Will do so!

File: yellowbrick/features/pca.py (hunk 4 above)
Comment (naresh-bachwani) [reply yellowbrick/features/pca.py]: OK. Will do so!

File: yellowbrick/features/pca.py (hunk 3 above)
Comment (naresh-bachwani) [reply yellowbrick/features/pca.py]: OK. Will do so!

File: yellowbrick/features/pca.py (hunk 2 above)
Comment (naresh-bachwani) [reply yellowbrick/features/pca.py]: Sure!

File: tests/test_features/test_pca.py (hunk 1 above)
Comment (naresh-bachwani) [reply tests/test_features/test_pca.py]: OK. I will construct a function for alpha test.

**Comment by lwgray** [comment]
@naresh-bachwani #806 was merge so whenever you are ready to open a new PR for PCA alpha addition, go ahead

**Comment by naresh-bachwani** [comment]
@lwgray, I will make a new PR once I have my test for alpha ready!!

**Comment by naresh-bachwani** [comment]
Hello, @lwgray @rebeccabilbro, 
What should be the default value for alpha in this case? If default alpha=0.75, then we can either increase tolerance or we need to make changes to baseline images. What should I do in this case? Please guide!

**Comment by lwgray** [comment]
@naresh-bachwani let’s move this discussion to the new PR even if not all the tests are passing.

**Comment by naresh-bachwani** [comment]
Ok! I am closing this PR and opening a new one.

**Pull request closed without merging**